File: project-euler/problem3.js
function largestPrimeFactor(number){
	var factors = [];
	// find the factors of the number first
	var n = number;
	while( n > 1 ){
		for(var i = 2; i < number; i++){
			if( n % i === 0){

				factors.push(i);
				n = n / i;
				break;
			}
		}
	}

	return factors[factors.length-1];
}

console.log(largestPrimeFactor(600851475143));

File: ruby-101/problem3.rb
def largest_prime_factor(number)
  factors = []
  n = number
  while n > 1
    (2..number-1).each do |i|
      if (n % i) === 0
        factors.push(i)
        n = n / i
        break
      end
    end
  end
  factors.last
end

total = largest_prime_factor(600851475143)
puts total.to_s
